galley/pkg/config/analysis/local/analyze_test.go

```diff
@@ -129,49 +132,86 @@ func TestAddRunningKubeSource(t *testing.T) {
 
 	mk := mock.NewKube()
 
-	sa := NewSourceAnalyzer(k8smeta.MustGet(), blankCombinedAnalyzer, "", nil, false)
+	sa := NewSourceAnalyzer(k8smeta.MustGet(), blankCombinedAnalyzer, "", "", nil, false)
+
+	sa.AddRunningKubeSource(mk)
+	g.Expect(sa.sources).To(HaveLen(2))
+}
+
+func TestAddRunningKubeSourceWithMeshCfg(t *testing.T) {
+	g := NewGomegaWithT(t)
+
+	istioNamespace := "istio-system"
+
+	cfg := &v1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: meshConfigMapName,
+		},
+		Data: map[string]string{
+			meshConfigMapKey: "",
+		},
+	}
+
+	mk := mock.NewKube()
+	client, err := mk.KubeClient()
+	if err != nil {
+		t.Fatalf("Error getting client for mock kube: %v", err)
+	}
+	if _, err := client.CoreV1().ConfigMaps(istioNamespace).Create(cfg); err != nil {
+		t.Fatalf("Error creating mesh config configmap: %v", err)
+	}
+
+	sa := NewSourceAnalyzer(k8smeta.MustGet(), blankCombinedAnalyzer, "", istioNamespace, nil, false)
 
 	sa.AddRunningKubeSource(mk)
-	g.Expect(sa.sources).To(HaveLen(1))
+	g.Expect(sa.sources).To(HaveLen(3))
```

Comment:
Out of curiosity, why is 3 correct here (the same comment applies to the rest of the tests)? Maybe a comment explaining why would be helpful, unless it's obvious.

Reply:
1. The default meshcfg source is always there, no matter what.
2. We added a meshcfg source for what's in the cluster configmap.
3. We also added a "regular" apiserver source for what's in the cluster.

The tests would probably be a bit stronger (as well as more obvious) if I was looking at the sources in a bit more detail instead of just counting them.

Reply:
Ok. If this is something that should be obvious to reviewers in this area, then leave it alone. Otherwise it might be nice to at least add a comment explaining why 3 is correct.

Not blocking on it.